include/jwt-cpp/jwt.h

```diff
@@ -1308,31 +1314,48 @@ namespace jwt {
 			 */
 			std::string sign(const std::string& data, std::error_code& ec) const {
 				ec.clear();
-				auto hash = this->generate_hash(data, ec);
-				if (ec) return {};
-
-				std::unique_ptr<RSA, decltype(&RSA_free)> key(EVP_PKEY_get1_RSA(pkey.get()), RSA_free);
-				if (!key) {
-					ec = error::signature_generation_error::get_key_failed;
+#ifdef JWT_OPENSSL_1_0_0
+				std::unique_ptr<EVP_MD_CTX, decltype(&EVP_MD_CTX_destroy)> md_ctx(EVP_MD_CTX_create(),
+																				  &EVP_MD_CTX_destroy);
+#else
+				std::unique_ptr<EVP_MD_CTX, decltype(&EVP_MD_CTX_free)> md_ctx(EVP_MD_CTX_new(), EVP_MD_CTX_free);
+#endif
+				if (!md_ctx) {
+					ec = error::signature_generation_error::create_context_failed;
 					return {};
 				}
-				const int size = RSA_size(key.get());
-
-				std::string padded(size, 0x00);
-				if (RSA_padding_add_PKCS1_PSS(key.get(), (unsigned char*)padded.data(),
-											  reinterpret_cast<const unsigned char*>(hash.data()), md(),
-											  -1) == 0) { // NOLINT(google-readability-casting) requires `const_cast`
+				EVP_PKEY_CTX* ctx = NULL;
```

Comment:
Really nitpicky, but this is a C++ library, so I think using nullptr instead of C style NULL would be more appropriate.